picard/solver.py

```diff
@@ -121,7 +125,21 @@ def picard(X, fun='tanh', n_components=None, ortho=True, whiten=True,
         Number of iterations taken to converge. This is
         returned only when return_n_iter is set to `True`.
     """
+    np.seterr(all='raise')
     random_state = check_random_state(random_state)
+
+    # Behaves like Fastica if ortho, standard infomax if not ortho
+    if extended is None:
+        if ortho:
+            extended = True
+        else:
+            extended = False
+
+    # Avoid possible overflows if the density is not tanh and extended is used.
+    if fun != 'tanh' and extended and not ortho:
+        warnings.warn('Using a different density than tanh for extended ica '
+                      'may result in wrong estimation and overflows')
```

Comment:
"may" ? why so? should it just be a warning?

Reply:
Its a bit technical, but in extended infomax, the non linearity is psi(x) = x+/-tanh(x), but you can imagine using any other function: psi(x) = x +/- f(x). What is important is that it remains a score function, e.g. is dominated by x^2 at infinity, and that x+f(x) corresponds to a  super gaussian and x - f(x) to a sub-gaussian density.

So the warning is just to say that if the user uses a custom function f, he should be cautious.